### ISSUE TITLE
Better console output of mysql database errors and implications

### ISSUE THREAD

**Issue by bluepuma77** [webanno/webanno#536]
Just trying to use Webanno for the first time. I would like to run it with Docker and have problems running it with a MySQL database. `SqlExceptionHelper - SQL Warning Code: -1100, SQLState: 02000`.

I tried [webanno3-docker](https://github.com/fkuhn/webanno3-docker) with an updated [Webanno WAR](https://github.com/webanno/webanno/releases/download/webanno-3.2.1/webanno-webapp-3.2.1.war). And I tried my own container with separated MySQL instance. The MySQL client connects without a problem, but WebAnno tells me `SqlExceptionHelper - SQL Warning Code: -1100, SQLState: 02000` and I get a message on the web `USE THIS INSTALLATION FOR TESTING ONLY -- AN EMBEDDED DATABASE IS NOT RECOMMENDED FOR PRODUCTION USE` - assuming WebAnno now switched to an internal database.

```
2017-06-20 19:57:12 INFO [SYSTEM] WebAnnoApplicationContextInitializer - 3.2.1 (2017-05-15 12:34:23, build 8a5a286910e6c34798ba1213d8db715ed2d50bcc)
2017-06-20 19:57:12 INFO [SYSTEM] WebAnnoApplicationContextInitializer - Authentication: database
2017-06-20 19:57:16 INFO [SYSTEM] RepositoryServiceDbData - Repository: /root/.webanno/repository
2017-06-20 19:57:17 INFO [SYSTEM] StandaloneShutdownDialog - Running in: tomcat
2017-06-20 19:57:17 INFO [SYSTEM] StandaloneShutdownDialog - Console: not available
2017-06-20 19:57:17 INFO [SYSTEM] StandaloneShutdownDialog - Headless: yes
2017-06-20 19:57:17 INFO [SYSTEM] StandaloneShutdownDialog - Running in server environment or from command line: disabling interactive shutdown dialog.
2017-06-20 19:57:18 WARN [SYSTEM] SqlExceptionHelper - SQL Warning Code: -1100, SQLState: 02000
2017-06-20 19:57:18 WARN [SYSTEM] SqlExceptionHelper - no data
2017-06-20 19:57:18 WARN [SYSTEM] SqlExceptionHelper - SQL Warning Code: -1100, SQLState: 02000
2017-06-20 19:57:18 WARN [SYSTEM] SqlExceptionHelper - no data
Jun 20, 2017 7:57:19 PM org.apache.catalina.startup.HostConfig deployWAR
INFO: Deployment of web application archive /opt/webanno/webapps/webanno.war has finished in 10,654 ms
Jun 20, 2017 7:57:19 PM org.apache.coyote.AbstractProtocol start
INFO: Starting ProtocolHandler ["http-bio-18080"]
Jun 20, 2017 7:57:19 PM org.apache.catalina.startup.Catalina start
INFO: Server startup in 10694 ms
```

It would be great to have better console output:
1. Print a descriptive error message instead of a code
2. Print infos like database `hostname` and `user` used to verify in case of error
3. Print info message if internal database is used

**Comment by reckart** [comment]
@bluepuma77 mind that https://github.com/fkuhn/webanno3-docker is not from us. You should probably file an issue in that repo. 


These errors only occur if the HSQLDB is used - I have no idea where they come from. However, they do not seem to be fatal and WebAnno works nevertheless.

```
2017-06-20 19:57:18 WARN [SYSTEM] SqlExceptionHelper - SQL Warning Code: -1100, SQLState: 02000
2017-06-20 19:57:18 WARN [SYSTEM] SqlExceptionHelper - no data
2017-06-20 19:57:18 WARN [SYSTEM] SqlExceptionHelper - SQL Warning Code: -1100, SQLState: 02000
2017-06-20 19:57:18 WARN [SYSTEM] SqlExceptionHelper - no data
```

There is an info message when the internal DB is used - you even mention it: `USE THIS INSTALLATION FOR TESTING ONLY -- AN EMBEDDED DATABASE IS NOT RECOMMENDED FOR PRODUCTION USE`

**Comment by reckart** [comment]
Anyway - it seems that WebAnno is not finding it's `settings.properties` file where the database config should be configured.

**Comment by bluepuma77** [comment]
Thanks for the quick response, where should the file `settings.properties` be?

**Comment by reckart** [comment]
It should be in the directory indicated by the `webanno.home` system property which seems to be set here: https://github.com/fkuhn/webanno3-docker/blob/master/webanno_initd

**Comment by bluepuma77** [comment]
Strange, it`s set to `/srv/webanno` and the file `settings.properties` exists.

```
database.dialect=org.hibernate.dialect.MySQL5InnoDBDialect
database.driver=com.mysql.jdbc.Driver
database.url=jdbc:mysql://mysql:3306/webanno
database.username=webanno
database.password=t0t4llYSecreT
```

**Comment by reckart** [comment]
I assume that the exported JAVA_OPTS doesn't survive the su command.

**Comment by bluepuma77** [comment]
Okay, I will just forget about the Docker version. Hoping for your official Docker version with 3.3.0 is coming soon.

Still some more debug output would be great. Like "_Loading settings from {path}_", "_Connecting to database {host} with user {user}_" and "_Using internal HSQLDB_".

**Comment by bluepuma77** [comment]
If you still want to try it with fkuhn: docker exec into container 
```
export JAVA_OPTS="-Djava.awt.headless=true -Xmx750m -XX:+UseConcMarkSweepGC -Dwebanno.home=/srv/webanno"
export CATALINA_BASE="/opt/webanno"
/usr/share/tomcat7/bin/shutdown.sh
/usr/share/tomcat7/bin/startup.sh
```

**Comment by reckart** [comment]
The location of the settings file is already logged **if there is a setting file**.

```
[SYSTEM] WebAnnoApplicationContextInitializer - Settings: /some/path/settings.properties
```

**Comment by reckart** [comment]
Actual relevant changes implemented in #543.
